experimental/jackPython/astMapping/main.py

```diff
@@ -26,79 +26,93 @@ def binOp_to_str(bop):
-    elif type(arg) == ast.Num:
-        return arg.n
-    elif type(arg) == ast.BinOp:
-        return binOp_to_str(arg)
-    else:
-        return ""
-
-
+def module(node):
+    gast = {"type": "root"}
+    gast["body"] = node_to_gast(node.body)
+    return gast
 
 """
-takes list of arguments in python ast and converts them to a list of
-strings
+takes a node that represents a list of nodes.
+returns a list of gast
 """
-def pyargs_to_strlist(args):
-    out = []
-    for arg in args:
-        out.append(pyarg_to_str(arg))
-            
-    return out
+def node_list(node):
+    gast_list = []
+    for i in range(0, len(node)):
+        gast_list.append(node_to_gast(node[i]))
+    
+    return gast_list
 
 """
 Takes python expressions and converts them to the generic
 ast format
 """
-def pyexpr_to_gast(node):
+def expr(node):
+    return node_to_gast(node.value)
+
+def call(node):
     gast = {}
-    if type(node.value) == ast.Call: # FIXME: should type of call be embedded in the gast?
-        if type(node.value.func) == ast.Name:
-            if node.value.func.id == "print":
-                gast["type"] = "logStatement"
-            else:
-                gast["type"] = "customStatement"
-            
-            # add arguments and return
-            gast["args"] =  pyargs_to_strlist(node.value.args)
-            return gast
+    gast["type"] = node_to_gast(node.func)
+    gast["args"] = node_to_gast(node.args)
+    return gast
 
 """
 takes python ast assigns and converts them to generic ast format
 note, that this assumes only a single assignment (i.e. x = 4)
 for now, it does not work for things link x,y = 4,5
 """
-def pyassign_to_gast(node):
+def assign(node):
     gast = {}
     gast["type"] = "varAssign"
-
-    # note that this does not work if there are multiple assignments in same line
-    firstTarget = node.targets[0]
-    gast["varId"] = node.targets[0].id
-
-    gast["varValue"] = pyarg_to_str(node.value) # note that need to make it work for arrays, tuples, etc.
+    gast["varId"] = node_to_gast(node.targets[0]) # FIXME: understand when targets won't be 0
+    gast["varValue"] = node_to_gast(node.value)
     return gast
 
-
-def py_to_gast(python_input_filename):
-    input_ast =  astor.code_to_ast.parse_file(python_input_filename)
-    print(astor.dump_tree(input_ast))
-    # TODO: can add more fields to the generic ast
-    gast = {"type": "root", "body": []}
-
-    # NOTE: with current implementation, it will just go until it sees something it recognizes
-    # eventually can implement nested structures
-    for node in input_ast.body:
-        if type(node) == ast.Expr:
-            gast["body"].append(pyexpr_to_gast(node))
-        if type(node) == ast.Assign:
-            gast["body"].append(pyassign_to_gast(node))
-    
-    return gast
+def name(node):
+    if node.id == "print":
+        return "logStatement"
+    else:
+        return "customStatement"
+                
+
+def node_to_gast(node):
```

Comment:
Is this name accurate? It seems like this is printing the string representation and not creating a gast?

Reply:
I think that this may be a design flaw in the way i wrote the function. Basically, the function will return a gast if a node that contains sub-nodes is passed to it. If the function is passed a base case, then it will simply return the string representation of the node.

For example:

- `node_to_gast(ast_generated_from_print("hello")) `will return the gast associated with `print("hello")`
- `node_to_gast(ast.String(s="hello"))` will return the string "hello"

This works since python is so flexible, but could definitely be a flaw. Does this make sense? May be easier to explain over video.
@Danish21